junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TempDirectoryParameterResolverTests.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+/**
+ * @since 5.9
+ */
+class TempDirectoryParameterResolverTests {
```

Comment:
Why are these tests not in `TempDirectoryCleanupTests`? Instead of mocking, I think they should follow the same strategy as the tests in the other class.

Reply:
This was based on TestReporterParameterResolverTests and was to test where @TempDir is not a File/Path. I've written parameter cleanup tests in TempDirectoryCleanupTests, and rewritten TempDirectoryParameterResolverTests to use AbstractJupiterTestEngineTests.